From 8fdc320b442c45f25044826715058ce3569f3ae1 Mon Sep 17 00:00:00 2001
From: Charles Giessen <charles@lunarg.com>
Date: Thu, 17 Sep 2020 16:31:16 -0600
Subject: [PATCH] Revert "loader: Return error if layer loading failed"

This reverts commit 49648d8fb398ac7d3d8e159939de060b8e759ecf.
Many layers were relying on silently failing to allow 32/64 bit versions
to work on linux. A later commit will properly check for failing layers.

Signed-off-by: Laurent Carlier <lordheavym@gmail.com>
---
 loader/loader.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/loader/loader.c b/loader/loader.c
index dd7f7d5b7..4d8865e13 100644
--- a/loader/loader.c
+++ b/loader/loader.c
@@ -5942,7 +5942,7 @@ VkResult loader_create_instance_chain(const VkInstanceCreateInfo *pCreateInfo, c
 
             lib_handle = loaderOpenLayerFile(inst, "instance", layer_prop);
             if (!lib_handle) {
-                return VK_ERROR_LAYER_NOT_PRESENT;
+                continue;
             }
 
             if (NULL == layer_prop->functions.negotiate_layer_interface) {
-- 
2.28.0

